hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/Iterators.scala

```diff
@@ -116,16 +116,16 @@ class LogFileIterator(logFiles: List[HoodieLogFile],
 
   // NOTE: This have to stay lazy to make sure it's initialized only at the point where it's
   //       going to be used, since we modify `logRecords` before that and therefore can't do it any earlier
-  private lazy val logRecordsIterator: Iterator[Option[HoodieRecord[_]]] =
+  private lazy val logRecordsIterator: Iterator[scala.Option[HoodieRecord[_]]] =
     logRecords.iterator.map {
-      case (_, record: HoodieSparkRecord) => Option(record)
-      case (_, _: HoodieEmptyRecord[_]) => Option.empty
+      case (_, record: HoodieSparkRecord) => scala.Option(record)
+      case (_, _: HoodieEmptyRecord[_]) => scala.Option.empty
       case (_, record) =>
         toScalaOption(record.toIndexedRecord(logFileReaderAvroSchema, payloadProps))
 
     }
 
-  protected def removeLogRecord(key: String): Option[HoodieRecord[_]] = logRecords.remove(key)
+  protected def removeLogRecord(key: String): scala.Option[HoodieRecord[_]] = logRecords.remove(key)
 
   protected def doHasNext: Boolean = hasNextInternal
 
```

Comment:
We should also call merger API here to check whether the record need to be dropped before load it as result of query.

Reply:
Hi @beyond1920, not quiet understand why we need to use merge here since the function tends to remove the record. Can you explain a bit?